constraints builder tests : adding tests + improvements

guilpier-code opened the pull request:
What's done here : 
- [x] split a big fixture in 2 independent parts
- [x] renaming
- [x] adding a test for week being not default
- [x] a bit of production code cleaning
- [ ]  what else ?

flomnes committed:
Add week number to additional constraint name

flomnes committed:
Fix tests

flomnes committed:
Add unit tests on the constraint namer

flomnes committed:
TODO

Commit:
constraint builder tests : split fixture in 2 parts

Commit:
constraint builder tests : split fixture in 2 parts completely

Commit:
constraint builder tests : mainly renaming

Commit:
constraint builder tests : make code more clear

Commit:
constraint builder tests : renaming and add a new test

Commit:
constraint builder tests : renaming

Commit:
Merge remote-tracking branch 'remotes/origin/develop' into fix/improve-constraints-builder-tests

Commit:
constraint builder tests : code formatting

flomnes commented on src/tests/src/solver/optimisation/constraints/constraints_builder.cpp:
```diff
@@ -141,33 +86,28 @@ struct BB
-                           .additionalConstraints = {addc3_netting},
-                           .clusterGlobalIndex = 2,
-                           .name = "cluster_3"};
+    PROPERTIES sts_propertes_1 = {.additionalConstraints = {addc1_withdrawal},
```
```suggestion
    PROPERTIES sts_properties_1 = {.additionalConstraints = {addc1_withdrawal},
```

guilpier-code replied:
As said in [this comment](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3305#discussion_r2630488977), it's corrected in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3306

guilpier-code commented on src/tests/src/solver/optimisation/constraints/constraints_builder.cpp:
```diff
@@ -141,33 +86,28 @@ struct BB
-                           .additionalConstraints = {addc3_netting},
-                           .clusterGlobalIndex = 2,
-                           .name = "cluster_3"};
+    PROPERTIES sts_propertes_1 = {.additionalConstraints = {addc1_withdrawal},
```
There are typos here : 
**sts_propertes_\<i\>** should be replaced with **sts_properties_\<i\>**

guilpier-code replied:
Was done in [next PR](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3306)

payetvin commented on src/solver/optimisation/opt_rename_problem.cpp:
```diff
@@ -411,26 +429,25 @@ void ConstraintNamer::BindingConstraintWeek(unsigned int constraint, const std::
-                                       constraint);
+    std::string location = LocationIdentifier(area_, AREA) + SEP + "ShortTermStorage" + "<"
+                           + sts_name + ">" + SEP + "Constraint" + "<" + constraint_name + ">";
+    std::string time = TimeIdentifier(timeStep_, WEEK);
```
Maybe refactor next lines with the code in ShortTermStorageCostVariation

guilpier-code replied:
This refactoring is precisely the subject of [PR 3306](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3306).

flomnes commented on src/tests/src/solver/optimisation/constraints/constraints_builder.cpp:
```diff
@@ -178,19 +118,53 @@ struct BB
+    std::vector<const char*> NomsDesPays = {"CountryA", "CountryB", "CountryC"};
+    uint32_t weekInTheYear = 1;
+
+    std::vector<double> Pi = std::vector(2 * nbTimeStepsOptim, 0.0);
+    std::vector<int> Colonne = std::vector(2 * nbTimeStepsOptim, 0);
+    int nombreDeContraintes = 0;
+    int nombreDeTermesDansLaMatriceDeContrainte = 0;
+    std::vector<int> IndicesDebutDeLigne = std::vector(4, 0);
+    std::vector<double> CoefficientsDeLaMatriceDesContraintes;
+    std::vector<int> IndicesColonnes;
+    std::vector<int> NombreDeTermesDesLignes = std::vector<int>(4, 0);
+    std::string Sens = std::string(4, '=');
```
```suggestion
    std::vector<double> Pi(2 * nbTimeStepsOptim, 0.0);
    std::vector<int> Colonne (2 * nbTimeStepsOptim, 0);
    int nombreDeContraintes = 0;
    int nombreDeTermesDansLaMatriceDeContrainte = 0;
    std::vector<int> IndicesDebutDeLigne(4, 0);
    std::vector<double> CoefficientsDeLaMatriceDesContraintes;
    std::vector<int> IndicesColonnes;
    std::vector<int> NombreDeTermesDesLignes(4, 0);
    std::string Sens(4, '=');
```

guilpier-code replied:
If we take the example of **Pi** : 
**Pi** a data member of a struct. Initializing **Pi** with : 
```c++
std::vector<double> Pi(2 * nbTimeStepsOptim, 0.0);
```
does not compile : it may be interpreted by compiler as a fonction declaration.

flomnes commented on src/tests/src/solver/optimisation/constraints/constraints_builder.cpp:
```diff
@@ -178,19 +118,53 @@ struct BB
+    int IncrementDAllocationMatriceDesContraintes = 10;
+    std::vector<CORRESPONDANCES_DES_VARIABLES> CorrespondanceVarNativesVarOptim;
+
+    std::vector<int> NumeroDeVariableStockFinal = std::vector<int>(10, -1);
```
```suggestion
    std::vector<int> NumeroDeVariableStockFinal(10, -1);
```

guilpier-code replied:
Same as above

flomnes commented on src/tests/src/solver/optimisation/constraints/constraints_builder.cpp:
```diff
@@ -178,19 +118,53 @@ struct BB
+    std::vector<std::vector<int>> NumeroDeVariableDeTrancheDeStock = std::vector<std::vector<int>>(
+      10,
+      std::vector<int>(5, -1));
+    std::vector<std::string> NomDesContraintes = std::vector<std::string>(100, "");
```
```suggestion
    std::vector<std::string> NomDesContraintes(100);
```

guilpier-code replied:
Same as above

flomnes commented on src/tests/src/solver/optimisation/constraints/constraints_builder.cpp:
```diff
@@ -178,19 +118,53 @@ struct BB
+    std::vector<CORRESPONDANCES_DES_VARIABLES> CorrespondanceVarNativesVarOptim;
+
+    std::vector<int> NumeroDeVariableStockFinal = std::vector<int>(10, -1);
+    std::vector<std::vector<int>> NumeroDeVariableDeTrancheDeStock = std::vector<std::vector<int>>(
+      10,
+      std::vector<int>(5, -1));
```
```suggestion
    std::vector<std::vector<int>> NumeroDeVariableDeTrancheDeStock(10, std::vector<int>(5, -1));
```

guilpier-code replied:
Same as above

flomnes reviewed (commented):
Style

guilpier-code committed:
Namers (variables and constraints) : improve readability (#3306)

Improves code readability in names.

What was done : 
- [x] class TargetVectorUpdater was removed (seemed overkill)
- [x] class **Namer** : 
   - [x] data members were made **private** instead of **public**
   - [x] many methods were made protected
- [x] some methods bodies were clarified
- [x] simplifications
- [x] renaming

---------

Co-authored-by: guilpier <guillaume.pierre_externe@rte-france.com>

flomnes committed:
Merge branch 'develop' into fix/improve-constraints-builder-tests
